### DISCUSSION TITLE
Initially encountered multiple bugs with the search solution - Primarily caused by not Republishing pages after updating their metadata

### DISCUSSION THREAD

**Discussion by spsys** [microsoft-search/pnp-modern-search#2316] [Q&A] [answered]
Hi, I am new to spfx and pnp search v4 but after installing in sandbox and live environment I have experienced several issues. 

Due to the high number of bugs I expect this must be locally related, particularly as others in the discussions do not appear to have the same level of obvious problems. Installation instructions appear very straight forward. Package is deployed without errors. Web parts are accessible and at a high level all appears to work as expected but when configuring each web part I am noticing multiple problems. 

Can anyone suggest anything obvious that could be causing this at tenant level perhaps? 

Apologies for the vague question, I wished to keep it high level initially. Happy to elaborate further if necessary....

Cheers

**Comment by wobba** [comment]
You don't call out what issues you are seeing. If you look at the network tab in the browser you can see if the search query API calls have errors or not. And are the errors on the SharePoint or Graph provider? If Graph, then you have to approve the Graph scopes for it to work. If SharePoint, then it could be a service issue on Microsoft's side of things.

**Reply by spsys** [reply]
Hi Mikael, thanks for taking the time to respond to my query. There doesn't appear to be any particular Network errors. Although I will continue to familiarise myself with the dev tools and investigate further.

Some examples of issues I am experiencing:

**Search Results & Search Filters:** 
Result (& thus filter) data is incorrect (possibly cached??). I have tagged pages with metadata i.e. a custom choice Language column. Initially a page was assigned a value "EN", I updated the column and changed the value to "English". Although the pages assigned Language is now "English", the returned search result is "EN". This is also visible for the RefinableString in the Debug view. It has been like this for >10 days. I have re-indexed the library several times and even re-indexed the site but this does not change.

**Search Results -> Layout Options -> Manage Card Fields:** 
- Cannot delete any Manage Card fields rows. Cursor is disabled on hover.
- PreviewImageUrl will not display after card fields have been modified, even after _Reset fields to default values_ (PreviewImageURL is working for the same data on another web part that hasn't had modified fields)

**Search Filters:** 
Changes are not saved when updating an existing item in Edit Filters, to edit an item it is necessary to delete item and re-add the item with the desired settings. 

I do not wish to take up much of your time but I really appreciate your help and guidance.

**Comment by spsys** [comment, marked as answer]
OK I just realised my primary issue (Search Results & Search Filters - results showing incorrect data) is happening if metadata was changed on a page since it was last published. I Unpublished & Republished each relevant page and the values in the Search Result and Filters web parts almost immediately reflected the expected values.

Despite some other quirks that I am becoming more familiar with, following the page "Republishing discovery" I am pleased to say that the solution is now working as expected. 

Great work from all involved in the development. I am happy to use modern search in various projects going forward
